com.unity.render-pipelines.core/Runtime/Textures/MSAASamples.cs

```diff
@@ -14,4 +18,52 @@ public enum MSAASamples
         /// <summary>MSAA 8X.</summary>
         MSAA8x = 8
     }
+
+    public static class MSAASamplesExtensions
+    {
+        public static FormatUsage AsFormatUsage(this MSAASamples samples)
+        {
+            switch (samples)
+            {
+                case MSAASamples.MSAA2x:
+                    return FormatUsage.MSAA2x;
+                case MSAASamples.MSAA4x:
+                    return FormatUsage.MSAA4x;
+                case MSAASamples.MSAA8x:
+                    return FormatUsage.MSAA8x;
+                default:
+                    return FormatUsage.Render;
+            }
+        }
+
+        public static MSAASamples Validate(this MSAASamples samples)
+        {
+            if (samples == MSAASamples.None)
+                return samples;
+
+            GraphicsFormat ldrFormat = SystemInfo.GetGraphicsFormat(DefaultFormat.LDR);
+            MSAASamples[] probe = {MSAASamples.MSAA2x, MSAASamples.MSAA4x, MSAASamples.MSAA8x};
+
+            int minSampleCount = 0;
+            int maxSampleCount = 0;
+
+            // Some GPUs might not support MSAA2x, so probe the supported range for clamping
```

Comment:
Maybe comment that it is for Metal API . Other platform don't have this problem

Reply:
I believe some APIs allow for the implementation to give you a different sample count than you asked for, but I'm not sure we can always probe for that.  Maybe Antti can clarify what this is defending against.

Reply:
Apple GPUs don't support MSAA 8x, but if we're checking the highest "generic format" MSAA range supported, you might as well check what is the lowest, because historically there has been GPUs outside of Metal domain that haven't supported 2x, but 4x only.